src/core/thread/dua_manager.cpp

```diff
@@ -75,23 +77,101 @@ void DuaManager::UpdateDomainUnicastAddress(BackboneRouter::Leader::DomainPrefix
 
     OT_ASSERT(prefix != NULL);
 
-    mDomainUnicastAddress.GetAddress().SetPrefix(prefix->mPrefix.mFields.m8, prefix->mLength);
     mDomainUnicastAddress.mPrefixLength = prefix->mLength;
+    mDomainUnicastAddress.GetAddress().Clear();
+    mDomainUnicastAddress.GetAddress().SetPrefix(prefix->mPrefix.mFields.m8, prefix->mLength);
 
-    if (Get<Utils::Slaac>().GenerateIid(mDomainUnicastAddress, NULL, 0, &mDadCounter) == OT_ERROR_NONE)
+    // Apply cached DUA Interface Identifier manually specified.
+    if (!mFixedDuaInterfaceIdentifier.IsUnspecified())
     {
-        Get<ThreadNetif>().AddUnicastAddress(mDomainUnicastAddress);
+        mDomainUnicastAddress.GetAddress().SetIid(mFixedDuaInterfaceIdentifier);
     }
     else
     {
-        mDomainUnicastAddress.GetAddress().Clear();
-        otLogWarnCore("Failed to generate valid DUA");
+        SuccessOrExit(GenerateDomainUnicastAddressIid());
     }
 
+    Get<ThreadNetif>().AddUnicastAddress(mDomainUnicastAddress);
+
 exit:
     return;
 }
 
+otError DuaManager::GenerateDomainUnicastAddressIid(void)
+{
+    otError error;
+
+    if ((error = Get<Utils::Slaac>().GenerateIid(mDomainUnicastAddress, NULL, 0, &mDadCounter)) == OT_ERROR_NONE)
+    {
+        otLogInfoCore("Generated DUA: %s", mDomainUnicastAddress.GetAddress().ToString().AsCString());
+    }
+    else
+    {
+        otLogWarnCore("Failed to generate valid DUA");
+    }
+
+    return error;
+}
+
+otError DuaManager::SetFixedDuaInterfaceIdentifier(const Ip6::InterfaceIdentifier *aIid)
+{
+    otError error;
+
+    // Clear the fixed Interface Identifier manually specified if there was.
+    if (aIid == NULL)
+    {
+        // Nothing to clear.
+        VerifyOrExit(!mFixedDuaInterfaceIdentifier.IsUnspecified(), error = OT_ERROR_NOT_FOUND);
+
+        // TODO - remove if (mDomainUnicastAddress.GetAddress().HasIid(mFixedDuaInterfaceIdentifier) &&
+        if (GetDomainUnicastAddress().HasIid(mFixedDuaInterfaceIdentifier) &&
+            Get<ThreadNetif>().IsUnicastAddress(GetDomainUnicastAddress()))
+        {
+            otLogInfoCore("Removed DUA IID: %s", mFixedDuaInterfaceIdentifier.ToString().AsCString());
+            mFixedDuaInterfaceIdentifier.Clear();
+            Get<ThreadNetif>().RemoveUnicastAddress(mDomainUnicastAddress);
+
+            if (GenerateDomainUnicastAddressIid() == OT_ERROR_NONE)
+            {
+                Get<ThreadNetif>().AddUnicastAddress(mDomainUnicastAddress);
```

Comment:
make sure mDomainUnicastAddress is cleared if `GenerateDomainUnicastAddressIid` failed?

Reply:
any specific reason to make the clear mandatory? any issue if not clear in your thought?

Reply:
I think we agreed to clear `DUA` whenever it's not added to Netif according to discuss https://github.com/openthread/openthread/pull/4862#discussion_r419208557
You can also check your comment for doing this: https://github.com/openthread/openthread/pull/4862/commits/fcfffd47d7d6dcf58759c8d0fd3844349ecd2d60

Reply:
as far as I can recall, clearing DUA also helped to resolve issues I mentioned in https://github.com/openthread/openthread/pull/4862#discussion_r418453423

Reply:
There might be some misunderstanding regarding the https://github.com/openthread/openthread/pull/4862#discussion_r419197006 
For all the three possible situations for the address enumerated there, I don't think any chance for your concern to happen. `fcfffd4` in #4862 is not a patch to fix something.